src/client/Microsoft.Identity.Client/Platforms/Features/RuntimeBroker/WamAdapters.cs

```diff
@@ -78,8 +79,9 @@ private enum ResponseStatus
             {
                 case ResponseStatus.UserCanceled:
                     logger.Error($"[WamBroker] {MsalError.AuthenticationCanceledError} {MsalErrorMessage.AuthenticationCanceled}");
-                    throw new MsalClientException(MsalError.AuthenticationCanceledError, MsalErrorMessage.AuthenticationCanceled);
-
+                    var clientEx = new MsalClientException(MsalError.AuthenticationCanceledError, MsalErrorMessage.AuthenticationCanceled);
+                    DecorateExceptionWithRuntimeErrorProperties(clientEx, authResult.Error);
+                    throw clientEx;
```

Comment:
Bit of a code smell. Easy to miss calling Decorate method. Can this be combined into one call? An internal constructor on exception class? Although maybe an issue with Interop reference... 
Or an extension method on Exception class? Something like 
```csharp
throw new MsalClientException(MsalError.AuthenticationCanceledError, MsalErrorMessage.AuthenticationCanceled).WithAdditionalErrorData(authResultError);
```